### PR TITLE
upgrade project to work with avalanchego v1.5.3

### PR DESCRIPTION
**Pull request by connorbode**
These are the changes I had to make to have the timestampvm & deploy on a local avash network running on avalanchego v1.5.3. Was able to complete all steps of the tutorial.

There are a number of `Timestamp` variables that were changed. It seems like the name is now used by the parent class causing a naming conflict. I used `VMTimestamp` instead but this was a really quick choice.

### CODE REVIEW

File: timestampvm/vm.go [hunk 1]
```diff
@@ -258,3 +259,24 @@ func (vm *VM) Connected(id ids.ShortID) error {
 func (vm *VM) Disconnected(id ids.ShortID) error {
 	return nil // noop
 }
+
+
```
Comment (ceyonur) [inline timestampvm/vm.go:263]: can we also put the same comments here like you did below?

**Review by ceyonur — CHANGES_REQUESTED** [review]
Good work. I was about to push similar changes with some differences. But since you were first comer let's merge yours :)

In short: avalanchego added a new `Time` field to core.Block. We can also use it here. No need to duplicate it.

File: timestampvm/block.go [hunk 2]
```diff
@@ -28,7 +28,7 @@ var (
 type Block struct {
 	*core.Block `serialize:"true"`
 	Data        [dataLen]byte `serialize:"true"`
-	Timestamp   int64         `serialize:"true"`
+	VMTimestamp int64         `serialize:"true"`
```
[the remark this answers is not in the document]
Comment (connorbode) [reply timestampvm/block.go]: Ok I've removed VM timestamp & have used the `core.Block` method (mostly just replaced everything with `Timestamp().Unix()`. Let me know if that's what you meant.

File: timestampvm/block.go [hunk 3]
```diff
@@ -55,13 +55,13 @@ func (b *Block) Verify() error {
 	}
 
 	// Ensure [b]'s timestamp is after its parent's timestamp.
-	if b.Timestamp < time.Unix(parent.Timestamp, 0).Unix() {
+	if b.VMTimestamp < time.Unix(parent.VMTimestamp, 0).Unix() {
```
[the remark this answers is not in the document]
Comment (connorbode) [reply timestampvm/block.go]: I believe this is resolved

File: timestampvm/service.go [hunk 4]
```diff
@@ -46,7 +46,7 @@ func (s *Service) ProposeBlock(_ *http.Request, args *ProposeBlockArgs, reply *P
 
 // APIBlock is the API representation of a block
 type APIBlock struct {
-	Timestamp json.Uint64 `json:"timestamp"` // Timestamp of most recent block
+	VMTimestamp json.Uint64 `json:"vmtimestamp"` // Timestamp of most recent block
```
[the remark this answers is not in the document]
Comment (connorbode) [reply timestampvm/service.go]: I believe this is resolved

File: timestampvm/vm.go (hunk 1 above)
Comment (connorbode) [reply timestampvm/vm.go]: Ok added

**Comment by connorbode** [comment]
Ok, I've resolved the change requests you made I believe

File: timestampvm/service.go [hunk 5]
```diff
@@ -46,7 +46,7 @@ func (s *Service) ProposeBlock(_ *http.Request, args *ProposeBlockArgs, reply *P
 
 // APIBlock is the API representation of a block
 type APIBlock struct {
-	Timestamp json.Uint64 `json:"timestamp"` // Timestamp of most recent block
+	Timestamp json.Uint64 `json:"vmtimestamp"` // Timestamp of most recent block
```
Comment (ceyonur) [inline timestampvm/service.go:49]: we can keep this json field as `timestamp`.

**Review by ceyonur — CHANGES_REQUESTED** [review]
Need to fix `vm.Initialize` methods in the `vm_test.go`. We can pass `nil` as the last parameter (for ` _ common.AppSender`).

Other than that LGTM.

**Review by ceyonur — CHANGES_REQUESTED** [review]
fixed tests. 

thanks for the PR!

note: sorry accidentally selected "Request for changes"

**Review by ceyonur — APPROVED** [review]

**Pull request merged**